jax/_src/export/serialization.py

```diff
@@ -246,30 +248,46 @@ def _serialize_pytreedef(
     children_vector_offset = _serialize_array(
         builder, _serialize_pytreedef, children
     )
+  custom_name = None
+  custom_auxdata = None
+  node_type = node_data and node_data[0]
 
   if node_data is None:  # leaf
     kind = ser_flatbuf.PyTreeDefKind.leaf
-  elif node_data[0] is type(None):
+  elif node_type is type(None):
```

Comment:
FYI you can also use `types.NoneType` here since 3.10: https://docs.python.org/3/library/types.html#types.NoneType.